dashboard/src/pages/api/run.ts

```diff
@@ -68,7 +68,7 @@ async function metaQuery(sql: string, params?: any[]): Promise<any> {
   if (!dbConns[db]?.isConnected) {
     // Trash this object and rebuild it
     delete dbConns[db];
-    dbConns[db] = await createConnection({ ...baseConnConfig, name: db, database: db });
+    dbConns[db] = await createConnection({ ...baseConnConfig, name: `${db}${Date.now()}`, database: db });
```

Comment:
I think we need a lot more than that here. The `dbConns` object is global between requests so a concurrency-induced race condition here could cause connections to be open but not exist in that object because it was overwritten with another `dbConns[db] = ...` (specifically if two requests to the same DB come in faster than `await createConnection` comes back, one will overwrite the other).

If we are giving them all different names, then we aren't planning on re-using the connections at all and this `dbConns` lookup table is an overcomplication we can drop.

Otherwise we should look into changing the type of `dbConns[db]` into a `Promise` and `await` retrieval of it, so the assignment into that data structure is blocking and other concurrent queries can see it and `await` for the connection themselves?

Reply:
why not just use a uuid?